records_mover/records/delimited/sniff.py

```diff
@@ -59,65 +52,154 @@ def sniff_hints_from_fileobjs(fileobjs: List[IO[bytes]],
+                    break
+            detector.close()
+            assert detector.result is not None
+            if 'encoding' in detector.result:
+                chardet_encoding = detector.result['encoding']
+                if chardet_encoding in hint_encoding_from_chardet:
+                    return hint_encoding_from_chardet[chardet_encoding]
+                else:
+                    logger.warning("Got unrecognized encoding from chardet "
+                                   f"sniffing: {detector.result}")
+                    return None
+            else:
+                logger.warning(f"Unable to sniff file encoding using chardet: {detector.result}")
+                return None
+    except OSError:
         logger.warning("Could not use chardet to detect encoding, as stream is not rewindable")
         return None
-    original_position = fileobj.tell()
-    fileobj.seek(0)
-    detector = chardet.UniversalDetector()
-    while True:
-        chunksize = 512
-        chunk = fileobj.read(chunksize)
-        detector.feed(chunk)
-        if detector.done or len(chunk) < chunksize:
-            break
-    detector.close()
-    fileobj.seek(original_position)
-    assert detector.result is not None
-    if 'encoding' in detector.result:
-        chardet_encoding = detector.result['encoding']
-        if chardet_encoding in hint_encoding_from_chardet:
-            return hint_encoding_from_chardet[chardet_encoding]
-        else:
-            logger.warning(f"Got unrecognized encoding from chardet sniffing: {detector.result}")
-            return None
+
+
+def csv_hints_from_python(fileobj: IO[bytes],
+                          record_terminator_hint: Optional[HintRecordTerminator],
+                          encoding_hint: HintEncoding) -> RecordsHints:
+    # https://docs.python.org/3/library/csv.html#csv.Sniffer
+    try:
+        with rewound_fileobj(fileobj) as fileobj:
+            # Sniffer tries to determine quotechar, doublequote,
+            # delimiter, skipinitialspace.  does not try to determine
+            # lineterminator.
+            # https://github.com/python/cpython/blob/master/Lib/csv.py#L165
+            python_encoding = python_encoding_from_hint[encoding_hint]
+            #
+            # TextIOWrapper can only handle standard newline types:
+            #
+            # https://docs.python.org/3/library/io.html#io.TextIOWrapper
+            #
+            if record_terminator_hint not in [None, '\n', '\r', '\r\n']:
+                logger.info("Unable to infer file with non-standard newlines "
+                            f"using Python csv.Sniffer {repr(record_terminator_hint)}.")
+                return {}
+            text_fileobj = io.TextIOWrapper(fileobj,
+                                            encoding=python_encoding,
+                                            newline=record_terminator_hint)
+            try:
+                # TODO: How to get 1024?  processing instructions?
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker